frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx

```diff
@@ -340,28 +354,26 @@ class NativeQueryEditor extends Component {
     // the completers when the editor mounts are the standard ones
     const standardCompleters = [...this._editor.completers];
 
-    this.swapCompleters = pos => {
-      this._editor.completers =
-        this.getSnippetNameAtCursor(pos) !== null
-          ? [{ getCompletions: this.getSnippetCompletions }]
-          : this.getCardSlugAtCursor(pos) !== null
-          ? [{ getCompletions: this.getCardTagCompletions }]
-          : standardCompleters;
-    };
+    this.nextCompleters = pos =>
+      this.getSnippetNameAtCursor(pos)
+        ? [{ getCompletions: this.getSnippetCompletions }]
+        : this.getCardTagNameAtCursor(pos)
+        ? [{ getCompletions: this.getCardTagCompletions }]
+        : standardCompleters;
```

Comment:
Chained ternaries are something we've had from early on, but in JS it is often considered an anti-pattern.

I've checked what direction we want to take and got one confirmation. Also it's something we've informally talked about in FE meetings: https://metaboat.slack.com/archives/C505ZNNH4/p1666099567587429

This may work well if extracted to a function.

Reply:
Done https://github.com/metabase/metabase/pull/25801/commits/d0367642c173366060eeb675c4c0ad03568d6738